contract_comm/evm.go

```diff
@@ -122,14 +122,23 @@ func QueryWithAddress(backend contracts.Backend, contractAddress common.Address,
 }
 
 func GetRegisteredAddress(registryId common.Hash, header *types.Header, state vm.StateDB) (common.Address, error) {
-	vmevm, err := createBackend(header, state)
+	vmevm, err := NewBackend(header, state)
 	if err != nil {
 		return common.ZeroAddress, err
 	}
 	return contracts.GetRegisteredAddress(vmevm, registryId)
 }
 
-func createBackend(header *types.Header, state vm.StateDB) (contracts.Backend, error) {
+func NewCachedBackend(header *types.Header, state vm.StateDB) (contracts.Backend, error) {
```

Comment:
This is `NewCachedBackend`, but the type is called `cacheBackend` rather than `cachedBackend`.  Personally, I would rename the type to `cachedBackend`.